packages/core/src/extensions/Blocks/nodes/Block.ts

```diff
@@ -173,264 +123,388 @@ export const Block = Node.create<IBlock>({
+            const { depth, startPos } = getBlockInfoFromPos(
+              state.doc,
+              state.selection.from
+            )!;
+
+            const selectionAtBlockStart =
+              state.selection.$anchor.parentOffset === 0;
+            const selectionEmpty =
+              state.selection.anchor === state.selection.head;
+            const blockAtDocStart = startPos === 2;
+
+            const posBetweenBlocks = startPos - 1;
+
+            if (
+              !blockAtDocStart &&
+              selectionAtBlockStart &&
+              selectionEmpty &&
+              depth === 2
+            ) {
+              return commands.BNMergeBlocks(posBetweenBlocks);
             }
+
             return false;
           }),
-        ({ chain }) =>
-          // we are at the start of a block at the root level. The user hits backspace to "merge it" to the end of the block above
-          //
-          // BlockA
-          // BlockB
-
-          // Becomes:
-
-          // BlockABlockB
-
-          chain()
-            .command(({ tr, state, dispatch }) => {
-              const isAtStartOfNode = tr.selection.$anchor.parentOffset === 0;
-              const anchor = tr.selection.$anchor;
-              const node = anchor.node(-1);
-              if (isAtStartOfNode && node.type.name === "block") {
-                if (node.childCount === 2) {
-                  // BlockB has children. We want to go from this:
-                  //
-                  // BlockA
-                  // BlockB
-                  //    BlockC
-                  //        BlockD
-                  //
-                  // to:
-                  //
-                  // BlockABlockB
-                  // BlockC
-                  //     BlockD
-
-                  // This parts moves the children of BlockB to the top level
-                  const startSecondChild = anchor.posAtIndex(1, -1) + 1; // start of blockgroup
-                  const endSecondChild = anchor.posAtIndex(2, -1) - 1;
-                  const range = state.doc
-                    .resolve(startSecondChild)
-                    .blockRange(state.doc.resolve(endSecondChild));
-
-                  if (dispatch) {
-                    tr.lift(range!, anchor.depth - 2);
-                  }
-                }
-                return true;
-              }
-              return false;
-            })
-            // use joinBackward to merge BlockB to BlockA (i.e.: turn it into BlockABlockB)
-            // The standard JoinBackward would break here, and would turn it into:
-            // BlockA
-            //     BlockB
-            //
-            // joinBackward has been patched with our custom version to fix this (see commands/joinBackward)
-            .joinBackward()
-            .run(),
-
-        () => commands.selectNodeBackward(), // (source: tiptap)
+        // Regular backspace behaviour.
+        () => commands.deleteSelection(),
```

Comment:
shouldn't deleteSelection still come first?

Reply:
Actually seems like this is redundant anyway, since I guess ProseMirror just uses the default backspace handler if none of your custom handlers return true.